### PR TITLE
Maven GitHub Action

### LINKED ISSUE #3: Build is not automated
we can make Maven build automated using github actions

### PR DESCRIPTION
**Pull request by h1alexbel**
closes #3

### CODE REVIEW

File: .github/workflows/maven.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        os: [ubuntu-20.04, windows-2022, macos-12]
+        java: [8, 11]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.java }}
+      - uses: actions/cache@v3
```
Comment (embano1) [inline .github/workflows/maven.yaml:23]: No need, see https://github.com/actions/setup-java

**Comment by embano1** [comment]
Branch protection should also be enabled, requiring this check to pass before merging

File: .github/workflows/maven.yaml (hunk 1 above)
Comment (h1alexbel) [reply .github/workflows/maven.yaml:23]: @embano1 should I just delete the actions/cache@v3 action from the `maven.yaml`?

**Comment by h1alexbel** [comment]
> Branch protection should also be enabled, requiring this check to pass before merging

@embano1 I believe it can be done with `add rule on branch` in the repository settings, I don't have enough permissions to add new rule

**Comment by embano1** [comment]
> @embano1 I believe it can be done with `add rule on branch` in the repository settings, I don't have enough permissions to add new rule

Yeah, you need permissions for that - or reach out to the team.

File: .github/workflows/maven.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: maven
+on:
```
Comment (embano1) [inline .github/workflows/maven.yaml:2]: nit: might want to include `permissions: {}` to scope them down to harden security (https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs)

Comment (h1alexbel) [reply .github/workflows/maven.yaml:3]: @embano1 done

**Review by hershalb — APPROVED** [review]

**Pull request merged**